cmd/xl-sets.go

```diff
@@ -556,34 +556,15 @@ func (s *xlSets) CopyObject(srcBucket, srcObject, destBucket, destObject string,
 		return srcSet.CopyObject(srcBucket, srcObject, destBucket, destObject, srcInfo)
 	}
 
-	// Initialize pipe.
-	pipeReader, pipeWriter := io.Pipe()
-
 	go func() {
-		if gerr := srcSet.GetObject(srcBucket, srcObject, 0, srcInfo.Size, pipeWriter, srcInfo.ETag); gerr != nil {
+		if gerr := srcSet.GetObject(srcBucket, srcObject, 0, srcInfo.Size, srcInfo.Writer, srcInfo.ETag); gerr != nil {
 			errorIf(gerr, "Unable to read %s of the object `%s/%s`.", srcBucket, srcObject)
-			pipeWriter.CloseWithError(toObjectErr(gerr, srcBucket, srcObject))
 			return
 		}
-		pipeWriter.Close() // Close writer explicitly signalling we wrote all data.
+		srcInfo.Writer.Close() // Close writer explicitly signalling we wrote all data.
```

Comment:
We have to check the error here. If the src object is encrypted Close() may fail because the object can be modified.

Reply:
Okay